server/blockchain/calypso/calypso_test.go

```diff
@@ -46,92 +48,108 @@ func TestExecute(t *testing.T) {
 }
 
 func TestCommand_AdvertiseSmc(t *testing.T) {
-	contract := NewContract([]byte{}, fakeAccess{})
+	contract := NewContract(fakeAccess{})
 
 	cmd := calypsoCommand{
 		Contract: &contract,
 	}
 
-	err := cmd.advertiseSmc(fake.NewSnapshot(), makeStep(t))
+	keyString := "dummy"
+	keyBytes := []byte(keyString)
+
+	snapshot := fake.NewSnapshot()
```

Comment:
why introduce (here and below) a snapshot variable, thereby executing all the checks on the same snapshot , instead of having separate snapshots per check ? 

This effectively couples all the checks together, if one misbehaves the error might show up much later.

I'd revert the changes here.